### DISCUSSION TITLE
Unable to get GitHub Pages with a custom domain with www subdomain and HTTPS to play nicely together

### DISCUSSION THREAD

**Discussion by britzl** [community/community#23185] [Other Feature Feedback, Questions, & Ideas]
<p>We’re in the process of migrating <a href="http://www.defold.com" rel="nofollow noopener">www.defold.com</a> from a proprietary site to GitHub Pages and in general everything is working well (Jekyll and static site generation works great!) BUT I’m unable to get our domain set up properly:</p>
<p>* Typing <a href="http://www.defold.com" rel="nofollow noopener">http://www.defold.com</a> ends up as <a href="https://defold.com" rel="nofollow noopener">https://defold.com</a> and site loads as expected</p>
<p>* Typing <a href="http://defold.com" rel="nofollow noopener">http://defold.com</a> ends up as <a href="https://defold.com" rel="nofollow noopener">https://defold.com</a> and site loads as expected</p>
<p>* Typing <a href="https://defold.com" rel="nofollow noopener">https://defold.com</a> works and site loads as expected</p>
<p>* Typing <a href="https://www.defold.com" rel="nofollow noopener">https://www.defold.com</a> DOES NOT WORK - Browser responds with “Your connection is not private”</p>
<p>Clicking the lock icon next to the adress on the three working examples above shows “Let’s Encrypt Authority” which seems to indicate that GitHub has generated SSL certificates as expected. The question is what goes wrong in the <a href="https://www.defold.com" rel="nofollow noopener">https://www.defold.com</a> case though? I’ve seen similar issues but I don’t really see what I’ve not configured correctly. This is how the site and DNS records are configured:</p>
<p>The site exists in organization repo <a href="http://github.com/defold/defold.github.io" class="inline-onebox-loading" rel="nofollow noopener">github.com/defold/defold.github.io</a> (private). In repo settings the Custom Domain field is set to <code>defold.com</code> and <code>Enforce HTTPS</code> is checked. No error is shown on the Settings page. The repo contains a CNAME file containing <code>defold.com</code> (which I believe gets auto generated by GitHub?).</p>
<p>We’re using AWS and Route 53 for DNS configuration. For <a href="http://defold.com" rel="nofollow noopener">defold.com</a> we have an IPv4 A record with value:</p>
<p>* 185.199.109.153<br>
* 185.199.108.153<br>
* 185.199.110.153<br>
* 185.199.111.153</p>
<p>For <a href="http://www.defold.com" rel="nofollow noopener">www.defold.com</a> we have a CNAME record with value:</p>
<p>* <a href="http://defold.github.io" rel="nofollow noopener">defold.github.io</a></p>
<p>What could be wrong?</p>

**Comment by lee-dohm** [comment]
<p>It looks like you got it working in the intervening time? (At least, clicking on the various links seems to work in all cases for me.) Can you confirm that things are working as expected for you?</p>
<p>Let us know.</p>

**Comment by britzl** [comment, marked as answer]
<p>Ah, yes, sorry. We got it working by rerouting DNS queries.</p>
<p><a href="/t5/forums/replypage/board-id/pages/message-id/2165/load-autosave/true">https://github.community/t5/forums/replypage/board-id/pages/message-id/2165/load-autosave/true</a></p>

**Comment by jakejarvis** [comment]
<p><a class="mention" href="/lee-dohm">@lee-dohm</a>  It seems like <a class="mention" href="/britzl">@britzl</a>’s using a band-aid fix by having Cloudfront handle the www site, is that correct? Definitely not an ideal solution…</p>
<p><a class="mention" href="/britzl">@britzl</a> Unfortunately there’s a long discussion about this at <a href="https://github.community/t/does-github-pages-support-https-for-www-and-subdomains/10360/35">https://github.community/t5/GitHub-Pages/Does-GitHub-Pages-Support-HTTPS-for-www-and-subdomains/m-p/34102</a> that’s been going on for almost a year — I’m dying to have this feature too, I hate having to depend on yet <em>another</em> service just to get HTTPS on a www redirect like the rest of the internet! :wink:</p>

**Comment by britzl** [comment]
<p>Jake! I completely agree with you and I really thought it was covered by GH Pages when I read the docs. I don’t want to rely on additional services either, but we had to release the new <a href="http://www.defold.com" rel="nofollow noopener">www.defold.com</a> site so we couldn’t really wait.</p>

**Comment by jakejarvis** [comment]
<p>Not judging you at all to be clear, I’m doing the same exact thing actually! :slight_smile:</p>
<p>Agree that they should make this much clearer in the docs if this isn’t going to be fixed anytime soon…</p>

**Comment by lhoupert** [comment]
<aside class="quote no-group" data-username="britzl" data-post="3" data-topic="10759" data-full="true">
<div class="title">
<div class="quote-controls"></div>
 britzl:</div>
<blockquote>
<p>Ah, yes, sorry. We got it working by rerouting DNS queries.</p>
<p><a href="/t5/forums/replypage/board-id/pages/message-id/2165/load-autosave/true">https://github.community/t5/forums/replypage/board-id/pages/message-id/2165/load-autosave/true</a></p>
</blockquote>
</aside>
<p>The link is broken, do you have another link?</p>
